woocommerce/woocommerce
# Fix removal of express payment method from state

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




With this PR, we are fixing the behavior of the REMOVE_AVAILABLE_EXPRESS_PAYMENT_METHOD reducer to correctly remove the specified express payment method.

Closes #44694.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable Stripe and Paypal express payment methods. 
2. Add a product to the Cart and visit the Checkout block page.
3. Notice there are 3 express checkout buttons.
4. Open console log and run `wp.data.dispatch( 'wc/store/payment').__internalRemoveAvailableExpressPaymentMethod('payment_request')` code to remove stripe's express payment method.
5. Confirm stripe express payment method is removed correctly.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Fixed a bug where express payment methods could not be removed from the wc/store/payment data store. 

#### Comment 

</details>
